src/Controller/Component/AuthComponent.php

```diff
@@ -267,7 +267,7 @@ public function authCheck(EventInterface $event): ?Response
         /** @var \Cake\Controller\Controller $controller */
         $controller = $event->getSubject();
 
-        $action = $controller->getRequest()->getParam('action');
+        $action = $controller->getRequest()->getParam('action', '');
         if (!$controller->isAction($action)) {
```

Comment:
How about `$action !== null &&` instead? Should be quite a bit faster than doing reflection on something that is already known to not exist?